src/gurt/shm_alloc.c

```diff
@@ -47,14 +48,47 @@ static uint64_t       page_size;
 
 extern __thread pid_t d_tid;
 
+#define INVALID_NUM_CORE (0xFFFFFFFF)
+static unsigned int
+get_cpu_core(void)
+{
+	hwloc_topology_t topology;
+	unsigned int     cpu_count;
+	int              depth;
+
+	/* Allocate, initialize, and perform topology detection */
+	hwloc_topology_init(&topology);
```

Comment:
My apologies to not have done it in the example, but the return code of the function calls to `hwloc_topology_init()` and `hwloc_topology_load()` should be checked.
From my understanding, the call to `hwloc_topology_destroy()` have to done only `hwloc_topology_init()` succeed.

Reply:
Thank you very much! Fixed as you suggested.